### PR TITLE
feat: first draft for sequencer

### PR DESCRIPTION
**Pull request by greged93**
Defines the main outline for the test sequencer.

Time spent on this PR: 0.1 day

Resolves: #103 

# Pull Request type

Please check the type of change your PR introduces:

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no API changes)
-   [ ] Build-related changes
-   [ ] Documentation content changes
-   [ ] Testing

# What is the new behavior?
Have a test sequencer crate in order to set the outline for the sequencer.

# Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: crates/sequencer/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+homepage.workspace = true
+repository.workspace = true
+readme.workspace = true
+license.workspace = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+# Starknet
+blockifier = { git = "https://github.com/starkware-libs/blockifier.git" }
```
Comment (ClementWalter) [inline crates/sequencer/Cargo.toml:16]: should we pin the blockifier?

File: crates/sequencer/src/sequencer.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+use crate::config::SequencerConfig;
+use blockifier::state::state_api::{State, StateReader};
+
+/// Sequencer is the main struct of the sequencer crate.
+/// Using a trait bound for the state allows for better
+/// speed, as the type of the state is known at compile time.
+pub struct Sequencer<S: State + StateReader> {
+    config: SequencerConfig,
```
Comment (ClementWalter) [inline crates/sequencer/src/sequencer.rs:8]: do we need to store the config?

Comment (ClementWalter) [reply crates/sequencer/src/sequencer.rs:8]: reminds me this https://www.thecodedmessage.com/posts/default-params/#scaling-defaults-in-rust-struct-update-syntax

File: crates/sequencer/Cargo.toml (hunk 1 above)
Comment (greged93) [reply crates/sequencer/Cargo.toml:16]: done, will pin on the latest tag

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**